hnn_core/network_builder.py

```diff
@@ -340,17 +340,6 @@ def _build(self):
         if _get_rank() == 0:
             print('[Done]')
 
-    def __enter__(self):
```

Comment:
Why aren't these needed anymore?

Reply:
I think if you go back in the history of `hnn-core` code, I had the idea to do something like this:

```py
with NetworkBuilder(net) as neuron_net:
    simulate_dipole(neuron_net)
```

so you ensure everything gets cleared when `__exit__` is called. I think Blake rightly argued that `NetworkBuilder` should not be visible to the user.